.github/CONTRIBUTING.md

```diff
@@ -1,27 +1,86 @@
+
+## Pull Requests
+
+If you wish to fix a bug or add new features, please discuss it with us in an issue first. If there's no issue, please create one using the [echarts issue helper](https://ecomfe.github.io/echarts-issue-helper/).
+
+## How to Debug ECharts
+
+The following steps help you to set up a developing environment for ECharts.
+
+### 1. Clone ECharts project
+
+If you wish to make pull requests, you should fork the ECharts project first. Otherwise, just clone it locally.
+
+```bash
+git clone git@github.com:apache/incubator-echarts.git
+```
+
+[ZRender](https://github.com/ecomfe/zrender) is the rendering library under the hood. You need to clone it along with ECharts.
+
+```bash
+git clone git@github.com:ecomfe/zrender.git
+```
+
+We assume these projects are downloaded at `~/workspace/echarts` and `~/workspace/zrender`. But their locations can be arbitrary.
+
+### 2. Install dependencies
+
+```bash
+cd ~/workspace/echarts
+npm install
+cd ~/workspace/zrender
+npm install
+```
+
+Sometimes, in order to fix an issue within echarts, changes have to be made inside the codebase of zrender. To test any changes to zrender locally you can use npm's [npm link](https://docs.npmjs.com/cli/link.html) feature, for example: 
+
+```bash
+cd ~/workspace/zrender
+npm link
+cd ~/workspace/echarts
+npm link
+```
+
+With this, you can see that `~/workspace/echarts/node_modules/zrender` is a link to `~/workspace/zrender`.
+
+### 3. Run and debug
+
+To build the ECharts project and watch source file changes (including ZRender project) to rebuild:
+
+```bash
+cd ~/workspace/echarts
+node build/build.js --watch
+```
+
+To build once:
+contributing-inspect.png
+```bash
+node build/build.js
+```
+
+Then, open the test cases under `~/workspace/echarts/test` in Web browser. You can add breakpoints under `src` directory. For example, in Chrome Inspect, it looks like:
+
+![Chrome inspect](../asset/contributing-inspect.png)
+
+### 4. Make a pull request
+
+Fork ECharts project into your own project. Checkout a branch from master branch named `fix-xxxx`, where xxxx is the issue id related. If there's no related issue, you need to create one in most cases to describe what's wrong or what new feature is required.
+
+If you are a committer of apache/incubator-echarts project, which means you have the write access to the project, you still need to push to a new branch (by `git push origin HEAD:refs/heads/fix-xxxx`) and use pull request to push your code. You cannot push code directly to `master` branch, otherwise it will be rejected by GitHub.
+
+Once your PR is merged, you will be asked to sign a PDF file called [Apache ICLA](https://www.apache.org/licenses/icla.pdf). This is required for all non-trival commits of the apache/incubator-echarts project, which means you don't have to sign it if you only changed some spelling in `README.md`.
```

Comment:
Shouldn't the CLA be signed before a PR is merged?

Reply:
I double checked the regulations, and found that ICLA is required only for new committers (who have write access to the project) rather than all contributors. So this is not necessary in fact.
> A signed ICLA is required to be on file before an individual is given commit rights to any ASF project. https://www.apache.org/licenses/
> If you are a brand new committer, you must first to complete and submit an Individual Contributor License Agreement (ICLA) before your committer account will be activated. https://www.apache.org/dev/new-committers-guide#cla

Reply:
Gotcha, I noticed this part is removed in #10121 which I think is a good thing but I wonder if we should consider putting this somewhere else so it's clear what the onboarding for new committers (write access to project) is? Or maybe Apache already has one, just wanted to check so it's not forgotten in the future

Reply:
@ClemMakesApps That's a good idea. I think we should provide this information along with other contribution-related information on our Web site.